src/App.tsx

```diff
@@ -1,24 +1,134 @@
-/* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, {
+  useCallback, useEffect, useMemo, useState,
+} from 'react';
+
 import { UserWarning } from './UserWarning';
+import { Header } from './components/Header';
+import { TodoList } from './components/TodoList';
+import { Footer } from './components/Footer';
+import { Notification } from './components/Notification';
+
+import { Select } from './types/Select';
+import { Todo } from './types/Todo';
+import { TodoData } from './types/TodoData';
+import { client, todoUrlEnd } from './utils/fetchClient';
+import {
+  createTodo,
+  deleteTodo,
+  changeTodo,
+} from './api/todos';
 
-const USER_ID = 0;
+const USER_ID = 10364;
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [select, setSelect] = useState(Select.All);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [isLoading, setIsLoading] = useState(false);
+
+  const activeTodos = todos.filter(({ completed }) => completed === false);
+  const completedTodos = todos.filter(({ completed }) => completed === true);
+
+  const getTodos = useCallback(async () => {
+    const data: Todo[] = await client.get(todoUrlEnd);
+
+    setTodos(data);
+  }, []);
+
+  useEffect(() => {
+    getTodos();
+  }, []);
+
+  const addTodo = useCallback(async (data: TodoData) => {
+    try {
+      setIsLoading(true);
+      await createTodo(data);
+    } catch {
+      setErrorMessage('Unable to add a todo');
+    } finally {
+      getTodos();
+      setIsLoading(false);
+    }
+  }, []);
+
+  const updateTodo = useCallback(async (
+    todoId: number,
+    data: string | boolean,
+  ) => {
+    try {
+      await changeTodo(todoId, data);
+    } catch {
+      setErrorMessage('Unable to update a todo');
+    } finally {
+      getTodos();
+      setIsLoading(false);
+    }
+  }, []);
+
+  const removeTodo = useCallback(async (id: number) => {
+    await deleteTodo(id);
+    getTodos();
+  }, []);
+
+  const handleSelectedTodos = useMemo(() => {
+    let visibleTodos: Todo[] = [...todos];
+
+    switch (select) {
+      case Select.Active:
+        visibleTodos = visibleTodos.filter(todo => todo.completed === false);
+        break;
+
+      case Select.Completed:
+        visibleTodos = visibleTodos.filter(todo => todo.completed === true);
+        break;
+
+      case Select.All:
+      default:
+        break;
+    }
+
+    return visibleTodos;
+  }, [todos, select]);
+
   if (!USER_ID) {
     return <UserWarning />;
   }
```

Comment:
I don't think that we need this component, it was necessary to see this message when we started work on this project, but now I think you can remove this if statement and `UserWarning` component